lib/payment_processor/clients/braintree/subscription.rb

```diff
@@ -6,6 +6,13 @@ def self.make_subscription(amount:, plan_id:, payment_method_token:)
           new(amount, plan_id, payment_method_token).subscribe
         end
 
+        def self.make_subscription_from_transaction(transaction)
+          amount = transaction...
+          plan_id = ENV['BRAINTREE_SUBSCRIPTION_PLAN_ID']
+          payment_method_token = transaction...
+          new(amount, plan_id, payment_method_token).subscribe
+        end
+
```

Comment:
We could just make a subscription with a specified `amount` - that's a value that gets added as a one-off payment on top of the subscription. The subscription quantity is specified with `price`.